portals/publisher/src/main/webapp/source/src/app/components/Apis/Details/Endpoints/GeneralConfiguration.jsx

```diff
@@ -109,14 +109,15 @@ function GeneralConfiguration(props) {
                         id: 'Apis.Details.Endpoints.GeneralConfiguration.Certificates.certificate.alias.exist',
                         defaultMessage: 'Adding Certificate Failed. Certificate Alias Exists.',
                     }));
-                } else if (err.response) {
+                } else if (err.response && err.response.body && err.response.body.description) {
                     Alert.error(err.response.body.description);
                 } else {
                     Alert.error(intl.formatMessage({
                         id: 'Apis.Details.Endpoints.GeneralConfiguration.Certificates.certificate.error',
                         defaultMessage: 'Something went wrong while adding the certificate.',
                     }));
                 }
+                return Promise.reject(err);
```

Comment:
Returning the error object directly in the catch callback is useful when you want to log the error or handle it in some other way, but don't want to interfere with the normal Promise chain behavior. Since we have already logged the error here, we can continue the promise chain with a new promise. So the approach is OK.